Datahub.Web/Elasticsearch/ElasticsearchService.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using Nest;
+using Elasticsearch.Net;
+using Elasticsearch.Net.Aws;
+using Datahub.Web.Models;
+
+namespace Datahub.Web.Elasticsearch
+{
+    public class ElasticsearchService : IElasticsearchService
+    {
+        private readonly ElasticClient _client;
+
+        /// <summary>
+        /// Initialises a new ElasticClient instance with support for localhost and AWS endpoints.
+        /// </summary>
+        public ElasticsearchService()
+        {
+            var builder = new UriBuilder();
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_SCHEME"))) 
+            {
+                builder.Scheme = Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_SCHEME");
```

Comment:
For production code I'd definitely avoid repeating the same string literal, which would also make it easier to read and DRYer. Plus GetEnvironmentVariable is probably a system call. Further down I refactored these kinds of calls into local variables just as I was reading it.